typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -68,23 +68,25 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
    */
   async deriveWarpRouteConfig(
     warpRouteAddress: Address,
-  ): Promise<HypTokenRouterConfig> {
+  ): Promise<DerivedTokenRouterConfig> {
     // Derive the config type
     const type = await this.deriveTokenType(warpRouteAddress);
-    const baseMetadata = await this.fetchMailboxClientConfig(warpRouteAddress);
+    const mailboxClientConfig = await this.fetchMailboxClientConfig(
+      warpRouteAddress,
+    );
     const tokenConfig = await this.fetchTokenConfig(type, warpRouteAddress);
     const remoteRouters = await this.fetchRemoteRouters(warpRouteAddress);
     const proxyAdmin = await this.fetchProxyAdminConfig(warpRouteAddress);
     const destinationGas = await this.fetchDestinationGas(warpRouteAddress);
 
     return {
-      ...baseMetadata,
+      ...mailboxClientConfig,
       ...tokenConfig,
       remoteRouters,
       proxyAdmin,
       destinationGas,
       type,
```

Comment:
This might be a bug. We already calculate a type for this TokenRouterConfig in `fetchTokenConfig()` and that returns a correct type. If you remove this line, the cast on the next line can also be removed.

Reply:
can prob remove `type` because it's being returned in `tokenConfig`